urllib3/exceptions.py

```diff
@@ -180,6 +180,14 @@ class SNIMissingWarning(HTTPWarning):
     pass
 
 
+class DependencyWarning(HTTPWarning):
```

Comment:
Hm doesn't really bother me,I feel better about something like BaseWarning (or HTTPWarning) than urllib3.Urllib3Warning.

I'm +0 to making it its own base warning though.